### PR TITLE
webcam script

### PR DESCRIPTION
**Pull request by tomnis**

### CODE REVIEW

File: capture_video.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+
+# set up the bridge
+b = Bridge("10.0.1.35")
+b.connect()
+
+frame_num = 0
+pick = []
+
+for frame in camera.capture_continuous(rawCapture, format="bgr", use_video_port=True):
```
Comment (tomnis) [inline capture_video.py]: just a test script that displays the captured video stream

File: hue/hue_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+from phue import Bridge
+
+
+class HueWrapper:
```
Comment (tomnis) [inline hue/hue_wrapper.py]: thin wrapper around phue library

File: main.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        return 45
+    # evening
+    elif hour <= 22:
+        return 600
+    else:
+        return 180
+
+
+
+def main():
```
Comment (tomnis) [inline main.py]: this is the main webcam script we run

File: main_only_motion.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    resolution= (640, 480)
+    camera = PiCamera()
+    camera.resolution = resolution
+    camera.framerate = 2
+    raw_capture = PiRGBArray(camera, size=resolution)
+    time.sleep(0.1)
+    return (camera, raw_capture)
+
+
+def scan(camera, capture, hue, strategy):
```
Comment (tomnis) [inline main_only_motion.py]: prototype script that only uses motion detector, i was just playing around with this for usage at night

File: optics/human_detector.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import cv2
+import imutils
+import time
+
+class HumanDetector:
```
Comment (tomnis) [inline optics/human_detector.py]: detects humans in the visible frame

File: optics/motion_detector.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import cv2
+
+class MotionDetector:
```
Comment (tomnis) [inline optics/motion_detector.py]: detects motion in the visible frame

File: main.py [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+            result = scan(camera, capture, hue, strategy)
+
+        finally:
+            # close the streams
+            camera.close()
+            capture.close()
+
+        if killer.kill_now:
+            print("received shutdown signal, closing video stream")
+            # close the camera and sleep
```
Comment (jayonhuh) [inline main.py]: todo: we may want to wake up before breaking out of the while loop as the pi might be sleeping

File: main_only_motion.py (hunk 4 above)
Comment (jayonhuh) [reply main_only_motion.py]: try harder bro

**Pull request merged**